pkg/storage/storageclass.go

```diff
@@ -0,0 +1,301 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "k8s.io/api/core/v1"
+	storageV1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClassBuilder provides struct for storageclass object containing
+// connection to the cluster and the storageclass definitions.
+type ClassBuilder struct {
+	// Storageclass definition. Used to create the storageclass object.
+	Definition *storageV1.StorageClass
+	// Created storageclass object
+	Object *storageV1.StorageClass
+	// Used in functions that define or mutate storageclass definition. errorMsg is processed before the storageclass
+	// object is created.
+	errorMsg  string
+	apiClient *clients.Settings
+}
+
+// AdditionalOptions additional options for storageclass object.
+type AdditionalOptions func(builder *ClassBuilder) (*ClassBuilder, error)
+
+// NewClassBuilder creates a new instance of ClassBuilder.
+func NewClassBuilder(apiClient *clients.Settings, name, provisioner string) *ClassBuilder {
+	glog.V(100).Infof(
+		"Initializing new storageclass structure with the following params: "+
+			"name: %s, provisioner: %s", name, provisioner)
+
+	builder := ClassBuilder{
+		apiClient: apiClient,
+		Definition: &storageV1.StorageClass{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+			Provisioner: provisioner,
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the storageclass is empty")
+
+		builder.errorMsg = "storageclass 'name' cannot be empty"
+	}
+
+	if provisioner == "" {
+		glog.V(100).Infof("The provisioner of the storageclass is empty")
```

Comment:
maybe set it yourself to "kubernetes.io/no-provisioner" instead.

Reply:
Not sure I follow. kubernetes.io/no-provisioner is only one possible provisioner

Reply:
"no-provisioner" I'd consider using it as default maybe.

Reply:
I personally think it should just be provided by the user. Otherwise the user needs some previous knowledge of what we have chosen the default to be which could be confusing.